policy/modules/system/systemd.te

```diff
@@ -1132,3 +1137,39 @@ files_var_filetrans(systemd_update_done_t, systemd_update_run_t, file, ".updated
 seutil_read_file_contexts(systemd_update_done_t)
 
 systemd_log_parse_environment(systemd_update_done_t)
+
+#########################################
+#
+# User session (systemd --user) local policy
+#
+
+allow systemd_user_session_type self:capability { dac_override sys_resource };
+allow systemd_user_session_type self:process setfscreate;
```

Comment:
This is fortunate by the way, this means that systemd --user will automatically transtition where needed as long as you have the fc-specs set up properly (this does not include socket activation though: setsockfscreate)

if does however (fortunately) do setsockcreate

Reply:
So i suppose the name-based auto type transition rules i mentioned earlier arent as important. As a matter of fact if you have your fc specs set up systemd should just create stuff with proper contexts automatically.

The thing is: we probably don't /shouldnt specify fc specs for content in /run/user/UID (i know i don't but i am not sure if refpolicy has any fc specs for content in /run/user/UID

Reply:
This is just the early structure for the policy. I didn't have any socket activating user services to test on the system I'm working on.

Reply:
> This is fortunate by the way, this means that systemd --user will automatically transtition where needed as long as you have the fc-specs set up properly (this does not include socket activation though: setsockfscreate)
> 
> if does however (fortunately) do setsockcreate

Hmm, actually i might be wrong here. Maybe it does do the setsockcreate properly. The thing here is that when you add fc-specs at runtime that might affect systemd, then you need to do a `systemctl (--user) daemon-reexec. You see, systemd loads the ffc-spec when it starts and then it caches them. So any specs you add at runtime will not be acknowledged by the running systemd instance